hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/TransitRegionStateProcedure.java

```diff
@@ -188,6 +222,27 @@ protected boolean waitInitialized(MasterProcedureEnv env) {
     return am.waitMetaLoaded(this) || am.waitMetaAssigned(this, getRegion());
   }
 
+  private void checkAndWaitForOriginalServer(ServerName lastHost)
+    throws ProcedureSuspendedException {
+    boolean isOnline = serverManager.findServerWithSameHostnamePortWithLock(lastHost) != null;
+    long retries = 0;
+    while (!isOnline && retries < forceRegionRetainmentRetries) {
+      try {
+        synchronized (this) {
```

Comment:
does it mean there may have multiple procedure tries to use this `checkAndWaitForOriginalServer` ?

Reply:
Not really, this would only be called by each TRSP individually. Removing this synchronized block on next commit.